packages/fault-detector/src/service.ts

```diff
@@ -151,15 +164,26 @@ export class FaultDetector extends BaseServiceV2<Options, Metrics, State> {
     // multiple requests of maximum 1000 blocks in the case that batchSize > 1000.
     let blocks: any[] = []
     for (let i = 0; i < batchSize; i += 1000) {
-      const provider = this.options
-        .l2RpcProvider as ethers.providers.JsonRpcProvider
-      blocks = blocks.concat(
-        await provider.send('eth_getBlockRange', [
+      let newBlocks: any[]
+      try {
+        newBlocks = await (
+          this.options.l2RpcProvider as ethers.providers.JsonRpcProvider
+        ).send('eth_getBlockRange', [
```

Comment:
I recommend we attempt to migrate away from `eth_getBlockRange` so that the services can work against infura/alchemy since that RPC is not provided, this can be achieved with the batch json rpc provider

Reply:
Alchemy definitely supports this RPC, not sure about infura.